packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,352 @@
+import fs from 'fs-extra';
+import inquirer from 'inquirer';
+import path from 'path';
+import { FunctionParameters } from 'amplify-function-plugin-interface';
+inquirer.registerPrompt('datetime', require('inquirer-datepicker'));
+import { CronBuilder } from '../utils/cronBuilder';
+const categoryName = 'function';
+
+var Days = {
+  MON: '1',
+  TUE: '2',
+  WED: '3',
+  THU: '4',
+  FRI: '5',
+  SAT: '6',
+  SUN: '7',
+};
+
+export async function scheduleWalkthrough(context: any, params: Partial<FunctionParameters>): Promise<Partial<FunctionParameters>> {
+  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
+  const resourceDirPath = path.join(projectBackendDirPath, categoryName, params.resourceName);
+  const cfnFileName = `${params.resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourceDirPath, cfnFileName);
+  let scheduleParams: Partial<FunctionParameters> = params;
+  if (params.cloudwatchEnabled === undefined || params.cloudwatchEnabled === 'false') {
+    if (await context.amplify.confirmPrompt.run('Do you want this function to be invoked on a schedule?', false)) {
```

Comment:
seems like the create and update flow should be separate methods? Then the create flow gets called from the function creation walkthrough and the update flow gets called from function update?

Reply:
This was decided in the design discussion. I will also ask Rene and Kaustav what they feel about it.